azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java

```diff
@@ -101,16 +104,45 @@ public void run() throws Exception {
 
     info(commands.size() + " commands to execute.");
     File[] propFiles = initPropsFiles();
-    Map<String, String> envVars = getEnvironmentVariables();
 
     // change krb5ccname env var so that each job execution gets its own cache
+    Map<String, String> envVars = getEnvironmentVariables();
     envVars.put(KRB5CCNAME, getKrb5ccname(jobProps));
 
+    // determine whether to run as Azkaban or run as effectiveUser
+    String executeAsUserBinaryPath = null;
+    String effectiveUser = null;
+    boolean isExecuteAsUser = determineExecuteAsUser(sysProps, jobProps);
+
+    if (isExecuteAsUser) {
+      String nativeLibFolder = sysProps.getString(NATIVE_LIB_FOLDER);
```

Comment:
If the concern is to make sure that the specified file exists, I have verified during testing that it will throw a FileNotFoundException further down the line, which is the expected behavior